notebooks/search-analysis/search_analysis.ipynb

```diff
@@ -0,0 +1,399 @@
+    "avg_score = avg_score.droplevel(0,1)\n",
+    "avg_score['h'] = avg_score['sem'] * t.ppf((1 + 0.95) / 2, avg_score['size'] - 1)\n",
+    "avg_score['lower'] = avg_score['mean'] - avg_score['h']\n",
+    "avg_score['upper'] = avg_score['mean'] + avg_score['h']\n",
+    "avg_score = avg_score.reset_index()\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "plt.style.use('ggplot')\n",
+    "plt.figure(figsize=(10,5))\n",
+    "fig, ax = plt.subplots(1, 1)\n",
+    "\n",
+    "plt.plot(avg_score['query_length'], avg_score['mean'], color='green')\n",
+    "ax.fill_between(avg_score['query_length'],avg_score['lower'],avg_score['upper'] , color='b', alpha=.1)\n",
+    "plt.xlabel(\"Query length\")\n",
+    "plt.xticks(rotation=45, ha=\"right\")\n",
+    "plt.ylabel(\"Average Precision @5\")\n",
+    "fig.set_size_inches(10,5)\n",
+    "plt.xlim((1,5))\n",
+    "plt.ylim((0,1))\n",
+    "plt.xticks(np.arange(1, 6, step=1))\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Average max scoring doc in top 5 \n",
+    "\n",
+    "vrel_rate = no_bestbets.groupby('search_term').max('score').reset_index()\n",
+    "vrel_rate = vrel_rate[['search_term','query_length','score']].groupby('query_length').agg(['mean','size','sem'])\n",
+    "vrel_rate = vrel_rate.droplevel(0,1)\n",
+    "vrel_rate['h'] = vrel_rate['sem'] * t.ppf((1 + 0.95) / 2, vrel_rate['size'] - 1)\n",
+    "vrel_rate['lower'] = vrel_rate['mean'] - vrel_rate['h']\n",
+    "vrel_rate['upper'] = vrel_rate['mean'] + vrel_rate['h']\n",
+    "vrel_rate = vrel_rate.reset_index()\n",
+    "vrel_rate\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "%matplotlib inline\n",
+    "plt.style.use('ggplot')\n",
+    "plt.figure(figsize=(10,5))\n",
+    "fig, ax = plt.subplots(1, 1)\n",
+    "\n",
+    "plt.plot(vrel_rate['query_length'], vrel_rate['mean'], color='green')\n",
+    "ax.fill_between(vrel_rate['query_length'],vrel_rate['lower'],vrel_rate['upper'] , color='b', alpha=.1)\n",
+    "plt.xlabel(\"Query length\")\n",
+    "plt.xticks(rotation=45, ha=\"right\")\n",
+    "plt.ylabel(\"Average max scoring doc\")\n",
+    "fig.set_size_inches(10,5)\n",
+    "plt.xlim(1,6)\n",
+    "plt.ylim(0,3)\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "no_bestbets.groupby(['session_id','search_term']).max('score').groupby('score').size()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Estimate the rate of returning a very relevant result within top 5 results\n",
+    "\n",
+    "from scipy.stats import beta\n",
+    "\n",
+    "a,b = 45,36\n",
```

Comment:
How did you determine the use of this numbers for the shape parameters of the beta distribution? I might need a stats refresher!

Reply:
not hardcoded any more!

a= number of times v relevant result returned 
b = number of times v relevant result not returned